File: 167. Two Sum II - Input array is sorted.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        
        if nums.count <= 0 {
            return []
        }
        
        var left = nums.startIndex
        var right = nums.endIndex - 1
        
        while left <= right {
            
            if nums[left] + nums[right] == target {
                return [left+1, right+1]
            }
            
            if nums[left] + nums[right] > target {
                right -= 1
                continue
            }
            
            if nums[left] + nums[right] < target {
                left += 1
                continue
            }
        }
        
        return []
    }
}

File: 167.Two Sum II - Input array is sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> result;
        
        if (numbers.size() == 0) {
            result.push_back(0);
            result.push_back(0);
            return result;
        }
        
        int index1 = 0;
        int index2 = numbers.size() - 1;
        
        while (index1 < index2) {
            int sum = numbers[index1] + numbers[index2];
            if (sum == target) {
                result.push_back(index1+1);
                result.push_back(index2+1);
                
                return result;
            } else if (sum < target) {
                index2--;
            } else {
                index1++;
            }
        }
        
        if (index1 >= index2) {
            result.push_back(0);
            result.push_back(0);
        }
        
        return result;
    }
};
